runtime/trampolines.m

```diff
@@ -1296,8 +1327,13 @@
 	if (array == NULL)
 		return NULL;
 
-	if (managed_class == NULL)
-		managed_class = mono_class_from_mono_type (managed_type);
+	MonoClass *mclass = NULL;
+	if (managed_class == NULL) {
+		mclass = mono_class_from_mono_type (managed_type);
+		managed_class = mclass;
+	}
+
+	xamarin_mono_object_release (&mclass);
```

Comment:
What happens here when `managed_class` is not null? What does `xamarin_mono_object_release` do with a `null`?